### DISCUSSION TITLE
Enhancement Request: UNDO for Folder Deletion

### DISCUSSION THREAD

**Discussion by jbutenhoff** [Kong/insomnia#6729] [Ideas]
I have found that when I open the context menu on a folder of requests and the context menu pops up, it regularly seems to also fire the clicked on whatever item in the context menu the mouse is positioned over when it first paints. Most of the items are not destructive, but sometimes it is Delete, and the entire folder is vaporized. Would very much apricate if there was either an UNDO or a confirmation before delete. This can be quite painful when there are many items within the folder & any documentation is lost. The root cause may be my pointer/mouse hardware, but the risk of data loss is real, regardless.

**Comment by siddhant-grfrs** [comment]
> I open the context menu on a folder of requests and the context menu pops up, it regularly seems to also fire the clicked on whatever item in the context menu the mouse is positioned over when it first paints. 

This is the real problem. It is such an annoyance.

**Reply by liyunccc** [reply]
yes, I often click on the delete button. It makes me angry.

**Comment by sunny-chung** [comment]
Actually, most of the time we do not intent to click the delete button. It was "clicked" due to a bug in Insomnia. However, they regard this is an enhancement and leave it as is, although receiving lots of criticism.

#6487 #6527 #6686

After deleting a folder with tens of requests unintentionally, I abandoned Insomnia and created another HTTP client software as a replacement.
